frontend/src/data/variables/AcsHealthInsuranceProvider.ts

```diff
@@ -0,0 +1,140 @@
+    ]);
+  }
+
+  getDatasetId(breakdowns: Breakdowns): string {
+    if (breakdowns.hasOnlySex()) {
+      return breakdowns.geography === "county"
+        ? "acs_health_insurance-health_insurance_by_sex_county"
+        : "acs_health_insurance-health_insurance_by_sex_state";
+    }
+
+    if (breakdowns.hasOnlyRace()) {
+      return breakdowns.geography === "county"
+        ? "acs_health_insurance-health_insurance_by_race_county"
+        : "acs_health_insurance-health_insurance_by_race_state";
+    }
+
+    // Age only breakdown is not supported yet, due to the dataset not being
+    // Aggregated on the backend.
+    throw new Error("Not implemented");
+  }
+
+  async getDataInternal(
+    metricQuery: MetricQuery
+  ): Promise<MetricQueryResponse> {
+    const breakdowns = metricQuery.breakdowns;
+    const datasetId = this.getDatasetId(breakdowns);
+    const acsDataset = await getDataManager().loadDataset(datasetId);
+
+    let df = acsDataset.toDataFrame();
+
+    // If requested, filter geography by state or county level
+    // We apply the geo filter right away to reduce subsequent calculation times
+    df = this.mergeStateCountyFips(df);
+    df = this.filterByGeo(df, breakdowns);
+    df = this.renameGeoColumns(df, breakdowns);
+    df = df.renameSeries({ race: "race_and_ethnicity" });
+    df = df.parseInts([
+      "with_health_insurance",
+      "without_health_insurance",
+      "total_health_insurance",
+    ]);
+
+    if (breakdowns.geography === "national") {
+      df = df
+        .pivot(
+          [
+            breakdowns.demographicBreakdowns.race_and_ethnicity.columnName,
+            breakdowns.demographicBreakdowns.sex.columnName,
+          ],
+          {
+            fips: (series) => USA_FIPS,
+            fips_name: (series) => USA_DISPLAY_NAME,
+            with_health_insurance: (series) => series.sum(),
+            without_health_insurance: (series) => series.sum(),
+            total_health_insurance: (series) => series.sum(),
+          }
+        )
+        .resetIndex();
+    } else {
+      df = df.pivot(
+        ["fips", "fips_name"]
+          .concat(
+            breakdowns.demographicBreakdowns.age.enabled
+              ? [breakdowns.demographicBreakdowns.age.columnName]
+              : []
+          )
+          .concat(
+            breakdowns.demographicBreakdowns.sex.enabled
+              ? [breakdowns.demographicBreakdowns.sex.columnName]
+              : []
+          )
+          .concat(
+            breakdowns.demographicBreakdowns.race_and_ethnicity.enabled
+              ? [breakdowns.demographicBreakdowns.race_and_ethnicity.columnName]
+              : []
+          ),
+        {
+          with_health_insurance: (series) => series.sum(),
+          without_health_insurance: (series) => series.sum(),
+          total_health_insurance: (series) => series.sum(),
+        }
+      );
+    }
+
+    let totalPivot: { [key: string]: (series: ISeries) => any } = {
```

Comment:
Oof this was confusing to read- are you doing this so that you can support multiple breakdowns? I'm inclined to not complicated things and just use getSoleDemographicBreakdown() again here like in AcsPopulationProvider.

I think it's confusing to add more complex code now to support invalid use cases (multiple demographic breakdowns) just because they may come in the future.